galite-core/src/main/kotlin/org/kopi/galite/form/VBlock.kt

```diff
@@ -1763,9 +1765,65 @@ abstract class VBlock(var form: VForm) : VConstants, DBContextHandler, ActionHan
   }
 
   protected fun fetchLookup(table: Int, currentField: VField) {
-    TODO()
-  }
 
+    val tab = object : Table(tables!![table]) {}
+    val columns = mutableListOf<Column<String>>()
+    val conditions = mutableListOf<Op<Boolean>>()
+
+    for (i in fields.indices) {
+      val f = fields[i]
+
+      if (f !== currentField && f.lookupColumn(table) != null && !f.isLookupKey(table)) {
+        f.setNull(activeRecord)
+      }
+    }
+
+    for (i in fields.indices) {
+      val column = fields[i].lookupColumn(table)
+      val col = Column<String>(tab, column!!, VarCharColumnType())
+
+
+      if (column != null) {
+        columns.add(col)
+      }
+      if (fields[i] == currentField || fields[i].isLookupKey(table)) {
+        val condition = fields[i].getSearchCondition()
+
+        val cond = object : Op<Boolean>() {
+          override fun toQueryBuilder(queryBuilder: QueryBuilder) {
+            TODO("Not yet implemented")
+          }
+        }
+
+        if (condition == null || !condition.startsWith("= ")) {
+          conditions.add(cond)
+        }
+      }
+    }
+
+    try {
+      transaction {
+        var t: Op<Boolean> = conditions[0]
+        conditions.minusElement(t).forEach {
+          t = t and it
```

Comment:
rename "t" to condition or something else .
Also check if you might already get the list of conditions with the "and"

Reply:
Could be replaced with conditions.compoundAnd()